src/v/kafka/server/offset_translator.h

```diff
@@ -53,10 +53,21 @@ class offset_translator {
 private:
     int64_t delta(model::offset o) const {
         auto it = _cfg_mgr.lower_bound(o);
-        if (it != _cfg_mgr.begin()) {
-            --it;
+        if (it == _cfg_mgr.begin()) {
+            if (it->first == o) {
```

Comment:
I wonder if `std::upper_bound` would be more appropriate here - IIUC we can get rid of the if.

Reply:
Or maybe I'm wrong. What is the kafka offset of the configuration batch anyway?

Reply:
I simplified this with `upper_bound`